scripts/pretty-printers/gdb/install.py

```diff
@@ -1,63 +1,60 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import os
 
-# This is the code that should be copied if you're applying the changes by hand.
-# Replace {0} with the path to this folder.
-file_contents = """
-python
-import sys
-import os
-
-pretty_printer_folder = '{0}'
-if os.path.exists(pretty_printer_folder):
-    sys.path.insert(1, pretty_printer_folder)
-    from pretty_printers import load_cbmc_printers
-    load_cbmc_printers()
-end
-"""
-
 
 def create_gdbinit_file():
     """
-    Add or append to a .gdbinit file the python code to set-up cbmc pretty-printers.
+    Create and insert into a .gdbinit file the python code to set-up cbmc pretty-printers.
     """
 
     print("Attempting to enable cbmc-specific pretty-printers.")
 
     home_folder = os.path.expanduser("~")
     if not home_folder:
-        print(home_folder + " is an invalid home folder, please manually create a .gdbinit file and apply the code.")
+        print(home_folder + " is an invalid home folder, can't auto-configure .gdbinit.")
         return
 
+    # This is the code that should be copied if you're applying the changes by hand.
+    gdb_directory = os.path.dirname(os.path.abspath(__file__))
+    code_block = \
+        [
+            "cbmc_printers_folder = '{0}'".format(gdb_directory),
+            "if os.path.exists(cbmc_printers_folder):",
+            "    sys.path.insert(1, cbmc_printers_folder)",
+            "    from pretty_printers import load_cbmc_printers",
+            "    load_cbmc_printers()",
+        ]
+
     gdbinit_file = os.path.join(home_folder, ".gdbinit")
-    file_write_mode = 'w'
+    lines = []
     if os.path.exists(gdbinit_file):
-        print(".gdbinit file exists at " + gdbinit_file + "."
-              " Please type 'y' if you want to append the pretty-printer commands to it. Press any other key to exit.")
-        while True:
-            choice = input().lower()
-            if choice == 'y':
-                file_write_mode = 'a'
-                break
-            else:
-                print("Not appending to file. Exiting.")
+        with open(gdbinit_file, 'r') as file:
+            lines = [ line.rstrip() for line in file ]
+    line_no = 0
+    imports = { "os", "sys" }
+    while line_no < len(lines):
```

Comment:
I was reluctant to put this in originally as if you had a .gdbinit file you already knew enough to apply this directly. You might want to save a `backup.gdbinit` somewhere just in case things are detected incorrectly and you make unwanted modifications to the file that can't be got back.